evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -1423,6 +1430,18 @@ where
         }
     }
 
+    fn check_block_timestamp(
+        builder: &mut CircuitBuilder<F, D>,
+        prev_timestamp: Target,
+        timestamp: Target,
+    ) {
+        // We check that timestamp > prev_timestamp.
+        // In other words, we range-check `diff = timestamp - prev_timestamp - 1`
+        // between 0 and 2ˆ32.
+        let diff = builder.sub(timestamp, prev_timestamp);
+        let diff = builder.add_const(diff, F::NEG_ONE);
```

Comment:
I don't think we should do a minus 1 here. Timestamp is UNIX based, in seconds if I'm correct. If a chain were to have a block time of < 1 sec this check would fail. We should probably allow for 0 diff.